src/app/components/colors/colors.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, forwardRef, Input, OnInit } from '@angular/core';
+import { ControlValueAccessor, FormBuilder, NG_VALUE_ACCESSOR, Validators } from '@angular/forms';
+
+import { colorsInterface } from 'src/app/core/interface/colorsInterface';
+import { requestStatus } from 'src/app/core/models/status.enum';
+
+
+@Component({
+  selector: 'app-colors',
+  templateUrl: './colors.component.html',
+  styleUrls: ['./colors.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => ColorsComponent),
+      multi: true,
+    },
+  ]
+})
+export class ColorsComponent implements ControlValueAccessor, OnInit {
+
+  requestStatus = requestStatus;
+
+  @Input() list!: colorsInterface;
+  disabled: boolean = false;
+
+  onChange = (value: any) => { };
+  onTouched = () => { };
+
+
+  constructor(public fb: FormBuilder) { }
+
+  ngOnInit(): void {
+    this.colorsForm.valueChanges.subscribe((value) => {
```

Comment:
we need to unsubscribe when this component will destroy

use take until strategy https://www.digitalocean.com/community/tutorials/angular-takeuntil-rxjs-unsubscribe